htlcswitch/link.go

```diff
@@ -2403,6 +2409,20 @@ func (l *channelLink) canSendHtlc(policy ForwardingPolicy,
 	payHash [32]byte, amt lnwire.MilliSatoshi, timeout uint32,
 	heightNow uint32) *LinkError {
 
+	// Check if the outgoing link has been disabled, this can occur due to
+	// manual user action. If disabled, an error is returned to the peer.
+	if policy.Disabled {
+		// As part of the returned error, we'll send our latest routing
+		// policy so the sending node obtains the most up-to-date data.
+		failure := l.createFailureWithUpdate(
+			func(upd *lnwire.ChannelUpdate) lnwire.FailureMessage {
+				return lnwire.NewChannelDisabled(uint16(
```

Comment:
Not sure if this is right. I think we use either `NewTemporaryChannelFailure` or `NewChannelDisabled` but not nesting them? `NewChannelDisabled` returns a `FailChannelDisabled` which has `CodeChannelDisabled` already.

Reply:
Agree